sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/Framework/AuthenticationEventRequest.cs

```diff
@@ -42,7 +42,7 @@ public async override Task<AuthenticationEventResponse> Completed()
             {
                 if (RequestStatus == RequestStatusType.Failed || RequestStatus == RequestStatusType.ValidationError)
                 {
-                    Response.MarkAsFailed(new Exception(string.IsNullOrEmpty(StatusMessage) ? AuthenticationEventResource.Ex_Gen_Failure : StatusMessage), true);
+                    Response.MarkAsFailed(new Exception(string.IsNullOrEmpty(StatusMessage) ? AuthenticationEventResource.Ex_Gen_Failure : StatusMessage), false);
```

Comment:
Our goal about using `ExceptionStatusCode` is so we don't have to pass true or false here. We should be able to look at the AuthenticationEventTriggerValidationException.ExceptionStatusCode in `Response.MarkAsFailed`

Reply:
This might force you to update Signature in a couple of spots. We should make sure the only exceptions that we throw inside of the library are AuthenticationEventTriggerException so we always know what the statuscode is from the exception and not figure it out on the fly